src/main/java/org/mule/extension/http/internal/request/HttpPollingSource.java

```diff
@@ -160,74 +169,122 @@ private ResponseValidator getResponseValidator() {
     return responseValidator != null ? responseValidator : defaultStatusCodeValidator;
   }
 
+  private void validateExpression(String exp) {
+    ValidationResult validation = isValidExpression(exp, expressionLanguage);
+    if (!validation.isSuccess()) {
+      throw new MuleRuntimeException(createStaticMessage(format("Invalid expression '%s' at HTTP Polling Source at %s - %s", exp,
+                                                                location.getRootContainerName(),
+                                                                validation.errorMessage().orElse(""))));
+    }
+
+  }
+
+  private void validateExpressions() {
+    validateExpression(requestBuilder.getRequestBody());
+    requestBuilder.getRequestHeaders().forEach(header -> validateExpression(header.getValue()));
+    requestBuilder.getRequestQueryParams().forEach(queryParam -> validateExpression(queryParam.getValue()));
+  }
+
   @Override
   protected void doStart() throws MuleException {
     LOGGER.debug("Starting HTTP Polling Source in {}", location.getRootContainerName());
     scheduler = schedulerService.ioScheduler();
     client = clientProvider.connect();
     httpRequester = createHttpRequester(false, muleContext);
-    resolvedUri = getResolvedUri();
+    validateExpressions();
```

Comment:
validations should be done at initialise, not start

Reply:
But.. how to? 
Sources don't have initialisation phase/method. I can't implement initialisable (SDK doesn't allow source to implement lifecycle interfaces)

Reply:
good point